### PR TITLE
fix: prevent splitting multibyte characters in `getPayload()`

### LINKED ISSUE #938: [BUG]: bodyparser in get-payload potentially buggy
### What happened?

I think @addaleax tweeted long time ago about this issue of bodyparser.

If you expect that the payload is using utf8 or utf16 or any other multybyte encoding, then you have to concat the buffers first and in the final step you can serialize the buffer to a string with the desired encoding. 

The issue will arise, when with some "luck" a multibyte character is split between two chunks.

### Versions

12.0.8

### Relevant log output

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by Uzlopak**
Resolves #938 

Resolves #ISSUE_NUMBER

----

### Before the change?

* 

### After the change?

* 

### Pull request checklist
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)

### Does this introduce a breaking change?

Please see our docs on [breaking changes](https://github.com/octokit/.github/blob/master/community/breaking_changes.md) to help!

- [ ] Yes
- [x] No

----

### CODE REVIEW

**Comment by wolfy1339** [comment]
Can you please add a test to make sure we don't break this behaviour in the future

**Comment by wolfy1339** [comment]
Please take the time to fill out the PR templates and use a more descriptive PR title

**Comment by Uzlopak** [comment]
@wolfy1339 
provided tests as requested. I also run a similar test against the original implementation and got for the multibyte test two broken characters.

**Comment by Uzlopak** [comment]
@wolfy1339 
adapted requested changes

File: test/integration/get-payload.test.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
+import EventEmitter from "node:events";
+import { getPayload } from "../../src/middleware/node/get-payload";
```
Comment (wolfy1339) [inline test/integration/get-payload.test.ts]:
```suggestion
import { getPayload } from "../../src/middleware/node/get-payload.ts";
```

**Review by wolfy1339 — APPROVED** [review]

**Pull request merged**